cmake-ide.el

```diff
@@ -124,6 +124,12 @@
   :group 'cmake-ide
   :safe #'stringp)
 
+(defcustom cmake-ide-cmake-command-flags
+  nil
+  "List of misc flags passed to the cmake invocation."
+  :group 'cmake-ide
+  :safe #'stringp)
```

Comment:
Hi, I have done something similar, and a string seems fine, you can set it to a bunch of option (like -DCMAKE_BUILD_TYPE=Debug -DOtherUsefullDef"
Then, I set it to "-DCMAKE_BUILD_TYPE=Debug" by default and use it to define the build directory.